src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponsesChatClient.cs

```diff
@@ -642,36 +633,31 @@ internal static FunctionTool ToResponseTool(AIFunctionDeclaration aiFunction, Ch
 
     internal static ImageGenerationTool ToImageResponseTool(HostedImageGenerationTool imageGenerationTool)
     {
-        ImageGenerationTool result = new();
-        ImageGenerationOptions? imageGenerationOptions = imageGenerationTool.Options;
-
-        // Model: Image generation model
-        result.Model = imageGenerationOptions?.ModelId;
-
-        // Size: Image dimensions (e.g., 1024x1024, 1024x1536)
-        if (imageGenerationOptions?.ImageSize is not null)
-        {
-            result.Size = new ImageGenerationToolSize(
-                imageGenerationOptions.ImageSize.Value.Width,
-                imageGenerationOptions.ImageSize.Value.Height);
-        }
+        ImageGenerationOptions? options = imageGenerationTool.Options;
 
-        // OutputFileFormat: File output format
-        if (imageGenerationOptions?.MediaType is not null)
+        return new()
         {
-            result.OutputFileFormat = imageGenerationOptions.MediaType switch
-            {
-                "image/png" => ImageGenerationToolOutputFileFormat.Png,
-                "image/jpeg" => ImageGenerationToolOutputFileFormat.Jpeg,
-                "image/webp" => ImageGenerationToolOutputFileFormat.Webp,
-                _ => null,
-            };
-        }
-
-        // PartialImageCount: Whether to return partial images during generation
-        result.PartialImageCount ??= imageGenerationOptions?.StreamingCount;
-
-        return result;
+            Background = imageGenerationTool.GetProperty<ImageGenerationToolBackground?>(nameof(ImageGenerationTool.Background)),
+            InputFidelity = imageGenerationTool.GetProperty<ImageGenerationToolInputFidelity?>(nameof(ImageGenerationTool.InputFidelity)),
+            InputImageMask = imageGenerationTool.GetProperty<ImageGenerationToolInputImageMask>(nameof(ImageGenerationTool.InputImageMask)),
```

Comment:
I may plumb API for this in the future, in which case this might be replaced.  It raises the question - for cases where we are plumbing in options do we still want to fallback to additional properties for those too (model, outputfileformat, etc)?

Reply:
Technically it'd be a breaking change not to, but I think it's reasonable if/when we promote things from additional properties to strongly-typed properties to treat it as a behavioral breaking change. I'd expect the number of folks depending on it to be minimal. But we could keep the fallback if desired; adds a little complexity, but doesn't hurt much.

Reply:
Sorry, I'm not following. What are you proposing we do?